### PR TITLE
Add option to allow SourceKit-LSP to prepare a target without lazy type checking

### PR DESCRIPTION
**Pull request by ahoppen**
rdar://130333458

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SKCore/SourceKitLSPOptions.swift [hunk 1]
```diff
@@ -178,6 +178,19 @@ public struct SourceKitLSPOptions: Sendable, Codable {
...
+  public enum BackgroundPreparationMode: String {
+    /// Build a target to prepare it
+    case build
+
+    /// Prepare a target without generating object files but do not do lazy type checking.
+    ///
+    /// This uses SwiftPM's `--experimental-prepare-for-indexing-no-lazy` flag.
+    case noLazy
+
+    /// Prepare a target without generating object files and the lik.
```
Comment (bnbarham) [inline Sources/SKCore/SourceKitLSPOptions.swift:190]:
```suggestion
    /// Prepare a target without generating object files.
```

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**